### PR TITLE
chore: vscode setting updates

### PR DESCRIPTION
**Pull request by alexanderbez**
- JSON prettyify/format
- Add proto formatting settings

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

### CODE REVIEW

**Review by ValarDragon — APPROVED** [review]
I feel like this won't work out of the box on Linux systems that don't have clang installed, but I believe this should be fine since vscode shouldn't error if it doesn't exist.

I'll see what happens on my machine without clang

**Comment by alexanderbez** [comment]
Yeah so you need clang-format installed, that is the caveat, but I think it's worth it for consistent proto formatting. WDYT?

**Comment by ValarDragon** [comment]
I'll install it locally (clang was actually on my setup todo list), I'm just hesistant about making it a required step vs a sub-component of some automated commands, since its another thing to add to the complexities of new dev's setting up a working environment. (It also gets fixed in every `make proto-format` call)

Adding it to the default settings.json is the least controversial thing though, since thats a "if its installed, use it in the way we prefer" situation, which is ideal!

File: .vscode/settings.json [hunk 1]
```diff
@@ -1,24 +1,28 @@
...
-        "editor.formatOnSave": true,
-        "editor.codeActionsOnSave": {
-          "source.organizeImports": true
-        }
-    },
-      "gopls": {
+  "tabnine.experimentalAutoImports": true,
+  "go.lintTool": "golangci-lint",
+  "go.formatTool": "gofumports",
+  "go.testOnSave": true,
```
Comment (p0mvn) [inline .vscode/settings.json:5]: I'm personally used to working with auto-save and just noticed the following:
image

Can we let everyone decide if they would like this enabled?

Comment (alexanderbez) [reply .vscode/settings.json:5]: Yes, wanna make a followup PR?

Comment (p0mvn) [reply .vscode/settings.json:5]: On it

**Pull request merged**